node/flatpak_node_generator/providers/npm.py

```diff
@@ -208,7 +206,7 @@ def __init__(
         self.all_lockfiles: Set[Path] = set()
         # Mapping of lockfiles to a dict of the Git source target paths and GitSource objects.
         self.git_sources: DefaultDict[
-            Path, Dict[Path, GitSource]
+            Path, Dict[Path, Tuple[str, GitSource]]
```

Comment:
should update the comment above to indicate what this is, since it's no longer *just* a GitSource. or maybe the value a dataclass like `NamedGitSource` or something instead, to make it clearer what the `str` is (and then you can just use attributes to get the innards, instead of having to do unpacking assignments)